### PR TITLE
Generate better output

### PR DESCRIPTION
**Pull request by Nyholm**
As @jderusse said before, maybe we dont need the traits. 

The purpose of the traits was to have non-auto-generated code.

### CODE REVIEW

File: src/S3/Result/CopyObjectOutputTrait.php [hunk 1]
```diff
@@ -8,18 +8,23 @@ trait CopyObjectOutputTrait
...
-        $this->VersionId = $this->xmlValueOrNull($data->VersionId);
-        $this->ServerSideEncryption = $this->xmlValueOrNull($data->ServerSideEncryption);
-        $this->SSECustomerAlgorithm = $this->xmlValueOrNull($data->SSECustomerAlgorithm);
-        $this->SSECustomerKeyMD5 = $this->xmlValueOrNull($data->SSECustomerKeyMD5);
-        $this->SSEKMSKeyId = $this->xmlValueOrNull($data->SSEKMSKeyId);
-        $this->SSEKMSEncryptionContext = $this->xmlValueOrNull($data->SSEKMSEncryptionContext);
-        $this->RequestCharged = $this->xmlValueOrNull($data->RequestCharged);
+        $this->CopyObjectResult = new CopyObjectResult([
+        'ETag' => $this->xmlValueOrNull($data->ETag, true),
+        'LastModified' => $this->xmlValueOrNull($data->LastModified, true),
```
Comment (Nyholm) [inline src/S3/Result/CopyObjectOutputTrait.php]: Is there any CS rules that we can make this look pretty?

File: src/S3/Result/CreateBucketOutputTrait.php [hunk 2]
```diff
@@ -8,9 +8,10 @@ trait CreateBucketOutputTrait
...
     protected function populateFromResponse(ResponseInterface $response): void
     {
-        $data = new \SimpleXMLElement($response->getContent(false));
+        $headers = $response->getHeaders(false);
+
+        $this->Location = $headers['Location'];
 
-        // TODO Verify correctness
-        $this->Location = $this->xmlValueOrNull($data->Location);
+        $data = new \SimpleXMLElement($response->getContent(false));
```
Comment (jderusse) [inline src/S3/Result/CreateBucketOutputTrait.php]: not used?

File: src/S3/Result/GetObjectOutput.php [hunk 3]
```diff
@@ -278,7 +278,7 @@ public function getObjectLockRetainUntilDate(): ?int
         $this->initialize();
 
         return $this->ObjectLockRetainUntilDate;
-    }
+    }CopyObjectOutputTrait
```
Comment (jderusse) [inline src/S3/Result/GetObjectOutput.php]: ?

Comment (Nyholm) [reply src/S3/Result/GetObjectOutput.php]: Oops

File: src/S3/Result/CreateBucketOutputTrait.php (hunk 2 above)
Comment (Nyholm) [reply src/S3/Result/CreateBucketOutputTrait.php]: Removed. Thank you

**Pull request merged**